packages/bloc/lib/src/bloc.dart

```diff
@@ -10,23 +9,26 @@ import '../bloc.dart';
 /// and transforms them into a `Stream` of `States` as output.
 /// {@endtemplate}
 abstract class Bloc<Event, State> extends Stream<State> implements Sink<Event> {
-  final PublishSubject<Event> _eventSubject = PublishSubject<Event>();
+  final _states = StreamController<State>();
+  final _events = StreamController<Event>.broadcast();
 
-  BehaviorSubject<State> _stateSubject;
+  State _currentState;
 
   /// Returns the current [state] of the [bloc].
-  State get state => _stateSubject.value;
+  State get state => _currentState;
 
   /// Returns the [state] before any `events` have been [add]ed.
   State get initialState;
 
   /// Returns whether the `Stream<State>` is a broadcast stream.
   @override
-  bool get isBroadcast => _stateSubject.isBroadcast;
+  bool get isBroadcast => _states.stream.isBroadcast
```

Comment:
I believe this should just be `_states.stream.isBroadcast` since this method is overridden for `Stream`.

Reply:
Hmm. Looks like `isBroadcast` value always equal `true` in previous version because `BehaviorSubject` contains `StreamController.broadcast()` as implementation which always return `isBroadcast =>true`. I changed this line to `isBroadcast => true`. @felangel What do you think?

Reply:
hmm I need to take a closer look at the BehaviorSubject and PublishSubject implementations but I would try to stay as consistent as possible with the existing implementations. I would also prefer to return `_state.stream.isBroadcast` rather than hardcoding the value in case the underlying implementation changes at some point. Thoughts?